01-front-end-basics/templating-handlebars/27-28/readme.md

```diff
@@ -19,110 +21,91 @@
+2) Using Handlebars, we create a structure we'd like to use for part of the page. Ours is for each student's link. This structure should be inside the script tag. Note the use of `{{ }}` to identify where data will go. 
 
-4. Compile your template in `js/app.js`. Handlebars has a function called `Handlebars.compile(source)`. The source is takes in is the html from inside your template script. Handlebars' `compile` function returns a function, which we'll save to a variable called `template`. This isn't a special varible name; it just reminds us what the function does. Later, we'll use the `template` function to  pass in data to render in the template part of the page.
+3) We make sure the data format we use matches what we have available to us. The example above uses the Spotify API. In this example, each student has a `first_name`, a `last_name`, and a `github_username`. 
 
-  ```js
-  // compile handlebars template
-  var source = $('#tracks-template').html();
-  var template = Handlebars.compile(source);
-  ```
 
-5. In the success handler for your AJAX call, pass the data that comes back from the API into your template function. The output of the template function is HTML that contains the data from the API. The last step is to append the HTML to the view.
+**Set up a factory line**
 
-  ```js
+4) Using `#each` in the template lets Handlebars know we'll want to repeat the structure inside the `#each` for every element of the array. 
 
-  function renderSpotifyData(data) {
-    // track results are in an array called `items`
-    // which is nested in the `tracks` object
-    var trackResults = data.tracks.items;
-    console.log(trackResults);
+### What is happening in `app.js`?
 
-    // ...
+**Select something to mold**
 
-    // pass in data to render in the template
-    var trackHtml = template({ tracks: trackResults });
+1) Using jQuery we select our `students-list-template` script element.
 
-    // append html to the view
-    $results.append(trackHtml);
-  }
-  ```
+**Make the mold**
+
+2) We give the element's inner html to Handlebars' `compile` function. Handlebars generates a `template` function for us (our html mold, if you wish).
+
+**Pour the plastic into the mold**
+
+3) The template from our html expects an array of students.  We send the template `data27.students` to use as `students` first. Then we use the template function again with `data28.students` as `students`.
+
+**Open our mold to find a casting!**
+
+4) Each time we call it, the template function returns a string of html that it has generated by mashing the data into the template.
+
+**Put the casting on display**
+
+5) We find select the element where we want the new HTML to appear and append the generated html strings to the page using jQuery.
```

Comment:
> original

5) We find select the element where we want the new HTML to appear and append the generated html strings to the page using jQuery.

> suggestion:

We use jQuery to select the element where we want the new HTML to appear and then append the generated html-strings to the page.